cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/common/DefaultStageConfigurer.java

```diff
@@ -81,6 +91,37 @@ public void setErrorSchema(@Nullable Schema errorSchema) {
     errorSchemaSet = true;
   }
 
+  @Override
+  public ValidationFailure addFailure(String message, @Nullable String correctiveAction) {
+    ValidationFailure failure = new ValidationFailure(message, correctiveAction);
+    failures.add(failure);
+    return failure;
```

Comment:
can't it have a cause? Like a property name and value?

Reply:
yes, it can, it will be used as below (will be part of next PR)
``` Java
 stageConfigurer.addFailure(configException.getMessage(),
                                     String.format("Provide valid value for config property '%s'.",
                                                   configException.getProperty()))
            .withPluginConfigCause(configException.getProperty());

Reply:
I see. 
That is not immediately clear. Perhaps add a code example to the Javadocs?